### DISCUSSION TITLE
How to make `WHERE IN` clause with composite unique key?

### DISCUSSION THREAD

**Discussion by Lui22** [prisma/prisma#21275] [Q&A]
### Question

I want to get all lessons which match my array of lessons.
```ts
const lessonsArray = [
  {
    date: new Date(),
    group_id: 1,
    subject_id: 1,
    start_time: '11:00'
  },
 ...
]
```

And I can't use `findMany`, i can't use `{ where: name: { in: lessonsArray } }`. What should I do?

### Information about Prisma Schema, Client Queries and Environment (optional)

My schema:
```prisma
model lessons {
  id          Int      @id @default(autoincrement())
  date        DateTime @db.Date
  start_time String
  group_id     Int?
  subject_id   Int?

  @@unique([date, group_id, subject_id, start_time], map: "name", name: "name")
}
```

**Comment by nurul3101** [comment]
Hello @Lui22 👋 

`@@unique([date, group_id, subject_id, start_time], map: "name", name: "name")` creates a composite unique index for the table `lessons`. It doesn't create a column in the table. You need to pass a column name in the `findMany` query.

Can you try doing something like this?

```ts
const lessonsArray = [
  {
    date: new Date(),
    group_id: 1,
    subject_id: 1,
    start_time: '11:00',
  },
  // ... other lessons
];

const orConditions = lessonsArray.map(lesson => ({
  date: lesson.date,
  group_id: lesson.group_id,
  subject_id: lesson.subject_id,
  start_time: lesson.start_time,
}));

const findResponse = await prisma.lessons.findMany({
  where: {
    OR: orConditions,
  },
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
